docker/archive/dest.go

```diff
@@ -19,17 +19,32 @@ func newImageDestination(ctx *types.SystemContext, ref archiveReference) (types.
 	if ref.destinationRef == nil {
 		return nil, errors.Errorf("docker-archive: destination reference not supplied (must be of form <path>:<reference:tag>)")
 	}
-	fh, err := os.OpenFile(ref.path, os.O_WRONLY|os.O_EXCL|os.O_CREATE, 0644)
+	// tries to open a file first to accomodate for load when using stdout
+	// as the file is created first then written to
+	// if the file doesn't exist then it creates a new file for exclusive use
+	// if the file exists, it checks to ensure the size is 0
```

Comment:
(To be precise, what matters is that the _image_ (config+layers) is the same. The _names_ may differ, both in the tag and in the repo (`busybox:latest` and `example.com/personalproject/myspecialcopy:today`), as long as they point to the same contents.)

#309 only allows _listing_ the various tags; but actually copying the images out still fails with the quoted error (see also the warning above; the code does not allow choosing which image to read).  (Just clarifying the current implementation state. The “load” path is fixable easily enough—we insist on seekable input, so it’s quite possible to choose one of the images; it’s the “save” part that is problematic.)

---

I guess short-term the easiest way to get the “save multiple” case working would be to create a temporary directory, copy all of the images in there one at a time using the `oci:` transport, and then tar it up. That bypasses the need to edit tar archives. Downsides would be:

- Extra code happening outside of containers/image, arguably working around its inflexible model
- OCI does not currently support multiple images _either_ ( #306 ), though that should be trivially fixable